lib/net.js

```diff
@@ -67,7 +67,13 @@ exports.connect = exports.createConnection = function() {
   args = normalizeConnectArgs(args);
   debug('createConnection', args);
   var s = new Socket(args[0]);
-  return Socket.prototype.connect.apply(s, args);
+
+  if (args.socketTimeout) {
+    s.setTimeout(args.socketTimeout);
+    return Socket.prototype.connect.apply(s, args);
+  } else {
```

Comment:
The `else` branch is unnecessary. You're returning the same thing in both branches.